src/plugins/discover/server/ui_settings.ts

```diff
@@ -163,4 +164,17 @@ export const uiSettings: Record<string, UiSettingsParams> = {
     category: ['discover'],
     schema: schema.arrayOf(schema.string()),
   },
+  [MODIFY_COLUMNS_ON_SWITCH]: {
+    name: i18n.translate('discover.advancedSettings.discover.modifyColumnsOnSwitchTitle', {
+      defaultMessage: 'Modify columns when index pattern is switched',
+    }),
+    value: true,
+    description: i18n.translate('discover.advancedSettings.discover.modifyColumnsOnSwitchText', {
+      defaultMessage:
+        'When switching index patterns, selected columns, that are not available in the ' +
+        'next index pattern, are removed.',
+    }),
+    category: ['discover'],
+    schema: schema.boolean(),
+  },
```

Comment:
dear @gchaps could you check this wording, many thx!
![Bildschirmfoto 2020-10-01 um 10 49 19](https://user-images.githubusercontent.com/463851/94806210-e0e26d00-03ed-11eb-8ec8-a5edbebdcae9.png)

Reply:
Modify columns when changing index patterns

Remove columns that not available in the new index pattern.

Reply:
@gchaps thx!